### PR TITLE
Update eclipse dependencies

### PR DESCRIPTION
**Pull request by candrews**
Update the `org.sonarsource.java:jdt org.eclipse.jdt:org.eclipse.jdt.core` and `org.eclipse.platform:*` dependencies.

I submit this PR for two reasons:
1. It's generally a good idea to keep dependencies up to date
2. The dependencies currently used by this project have pom's with `<modelVersion>4.0</modelVersion>` which causes problems with Gradle. The updated dependencies have pom's this that issue, using `<modelVersion>4.0.0</modelVersion>` . This problem was originally reported at https://github.com/eclipse-platform/eclipse.platform/issues/180

- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Unit tests are passing and you provided a unit test for your fix
- [ ] ITs should pass : To run ITs locally, checkout the [README](https://github.com/SonarSource/sonar-java/blob/master/README.md) of the project.
- [ ] If there is a [Jira](http://jira.sonarsource.com/browse/SONARJAVA) ticket available, please make your commits and pull request start with the ticket number (SONARJAVA-XXXX)

### CODE REVIEW

**Comment by candrews** [comment]
I created a topic about this issue at https://community.sonarsource.com/t/sonar-java-plugin-has-dependencies-with-invalid-poms/96334

**Comment by candrews** [comment]
@alban-auzeill @irina-batinic-sonarsource @Godin @Wohops @dorian-burihabwa-sonarsource @ADarko22 could you please take a look at this PR?

Thank you very much in advance.

**Review by deskpil0t — APPROVED** [review]
minor dependency changes

**Comment by johann-beleites-sonarsource** [comment]
For reference, we cannot currently easily upgrade our eclipse dependencies. You can find more details in [this ticket](https://sonarsource.atlassian.net/browse/SONARJAVA-4526) and [this community thread](https://community.sonarsource.com/t/sonar-java-plugin-has-dependencies-with-invalid-poms/96334).

TLDR: The main issue is that starting with 3.34, ECJ requires Java 17. We are in the process of deprecating and removing support for Java 11 for our users, however, this is not yet complete. We are very motivated to keep up to date with the current Java versions, for which we will need to upgrade ECJ, so rest assured that we are not forgetting about this. Once we officially drop support for Java 11 in our products, we can upgrade the ECJ dependency in sonar-java.
